### PR TITLE
Debounce refactor

### PR DESCRIPTION
**Pull request by chuckwagoncomputing**
Does this address all your concerns in #1801?

### CODE REVIEW

**Comment by chuckwagoncomputing** [comment]
My Discovery is on the blink for some reason, so I haven't tested this.

**Comment by chuckwagoncomputing** [comment]
My discovery started working, and this change does in fact break something.

**Comment by chuckwagoncomputing** [comment]
Well it looks like the bug I thought I had was just my discovery glitching.

File: firmware/hw_layer/debounce.cpp [hunk 1]
```diff
@@ -12,35 +12,41 @@
...
+    startConfiguration();
     initialized = true;
 }
 
-void ButtonDebounce::updateConfigurationList () {
+void ButtonDebounce::stopConfigurationList () {
+    ButtonDebounce *listItem = s_firstDebounce;
+    while (listItem != nullptr) {
+        listItem->stopConfiguration();
+        if (listItem->nextDebounce != nullptr) {
```
Comment (rusefillc) [inline firmware/hw_layer/debounce.cpp:38]: please remove one line here and three lines a bit below

"while (listItem != nullptr)" would have same exact effect, you effectively wrote the same code twice which is not useful

Comment (chuckwagoncomputing) [reply firmware/hw_layer/debounce.cpp]: done

**Comment by rusefillc** [comment]
this introduces new issue #1811

**Pull request merged**